packages/forms-web-app/src/controllers/examination/email.js

```diff
@@ -2,24 +2,61 @@ const {
 	routesConfig: {
 		examination: {
 			directory: examinationDirectory,
-			pages: { nameMyself, email }
+			pages: { email }
 		}
 	}
 } = require('../../routes/config');
 
+const config = require('../../config');
+const examinationSessionStorage = config?.sessionStorage?.examination;
+
 const pageData = {
-	backLinkUrl: `${examinationDirectory + nameMyself.route}`,
 	id: email.id,
 	pageTitle: email.name,
-	title: email.name
+	title: email.name,
+	hint: "We'll use your email address to confirm we've received your submission. We will not publish your email address."
 };
 
 const getEmail = async (req, res) => {
+	const requestSession = req.session;
+	const userEmail = requestSession?.[examinationSessionStorage.name]?.userEmail;
+	const currentView = requestSession?.currentView;
+	userEmail && (pageData.email = userEmail);
+
+	if (!requestSession || !currentView || !currentView?.route)
+		return res.status(404).render('error/not-found');
+
+	pageData.backLinkUrl = `${examinationDirectory + currentView.route}`;
 	res.render(email.view, pageData);
 };
 
 const postEmail = async (req, res) => {
-	res.render(email.view, pageData);
+	const { body = {} } = req;
+	const { errors = {}, errorSummary = [] } = body;
+	const requestSession = req?.session;
+	const currentView = requestSession?.currentView;
+	const examinationStorage = requestSession?.[examinationSessionStorage.name];
+
+	examinationStorage?.userEmail && (pageData.email = examinationStorage?.userEmail);
+
+	if (errors[email.id] || Object.keys(errors).length > 0) {
+		return res.render(email.view, {
+			...pageData,
+			errors,
+			errorSummary
+		});
+	}
+
+	if (!requestSession || !currentView) return res.status(404).render('error/not-found');
```

Comment:
As mentioned before, we would always have the 'requestSession' but what to check for the req.session.examination

Reply:
If someone sends a link to a friend for example and they open in the middle of the journey they should get a not found message as they don't have a session.

Reply:
They will always have a session (`req.session`) but we want to check if they have started the examination journey (`req.session.examination`)

Reply:
this was raised by Vimal while testing